.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+    types: [closed, opened]
+
+jobs:
+  dry-run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+      - run: npm ci
+      - run: npm test
+      - uses: JS-DevTools/npm-publish@v3
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
+          dry-run: true
+
+  publish:
+    needs: dry-run
+    if: github.event.pull_request.merged == true && (github.ref_name == 'main' || github.head_ref == 'main')
```

Comment:
since you don't have `push` triggers, and only `pull_request`, I think you can omit the `github.ref_name == 'main'`

Reply:
since I removed `branches: 'main'` in order for 'dry-run' to run on all branches I am relying on this to not publish on other branches. I suppose its not necessary for all branches to have this.

Reply:
I think I misunderstood, head_ref will be enough.